tests/rest/test_connection.py

```diff
@@ -513,6 +513,86 @@ def test_capabilities_caching(requests_mock):
     assert con.capabilities().api_version() == "1.0.0"
     assert m.call_count == 1
 
+def test_capabilities_caching_after_authenticate_basic(requests_mock):
+    user, pwd = "john262", "J0hndo3"
+
+    def get_capabilities(request, context):
+        endpoints = BASIC_ENDPOINTS
```

Comment:
FYI: the tests are failing because we are using BASIC_ENDPOINTS here, and appending directly to that, which causes leakage between tests.

So the fix for this is
```suggestion
        endpoints = BASIC_ENDPOINTS.copy()
```
(in both places)